### DISCUSSION TITLE
Automatic Translation

### DISCUSSION THREAD

**Discussion by tjthejuggler** [Docile-Alligator/Infinity-For-Reddit#853] [Duplicate Ideas] [answered]
I would love the ability to see posts and comments translated automatically without needing to use another app. Even if it means I need to put in my own google translate API or something else. I think it would be really cool for being able to get a sense of what sorts of things people are talking about in various parts of the world.

**Comment by History-exe** [comment]
Add a button to control whether to automatically translate globally. Or add a button for manual translation

1. sometimes the translation is not accurate. It will be awful to see the sentences on the full screen.

2. in addition, sometimes need to read the original text, and then see whether what you understand without translation is correct. To learn a foreign language.

**Comment by Wladefant** [comment, marked as answer]
Duplicate of #970 Please consider adding your ideas there

**Comment by Wladefant** [comment]
This is a very good function, that should be implemented, i really like the translate function in the youtube comments and would therefore recommend to implement something similar. This is an example in Youtube regarding the frontend:
![image](https://user-images.githubusercontent.com/79611093/178480418-e1931863-f3ed-4dc6-9501-031a1964e97f.png)

When we speak about the backend I would suggest we could use the Translation ML Kit from Google. This is an on-device Translation and supports more than 50 languages. These are offline models, that are also used in google translate offline mode, and therefore not as good as the google online translation.
If we want to make the implementation easy and intuitive, we could use the solution mentioned above and just follow this tutorial: https://developers.google.com/ml-kit/language/translation

Something worth mentioning is, that the models are all trained for translation from and to the English language, so the English language will always be used for translation.
Another thing is that we would need a language identification tool to make the translation enjoyable.

If anyone is interested in this feature please consider upvoting and let your comments, because there are also other solutions that i could present if other people are interested. 
@Docile-Alligator what do you think?
